Odroid h4 plus regression

mysteriousCzrsX opened the pull request.

mysteriousCzrsX committed:
self-tests/os-boot.robot: removed unused rte api reference

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
dasharo-compatibility/check-ethernet-ports-order.robot: added check to skip if the test shouldn't be run

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
platform-configs/odroid-h4-plus.robot: updated variables describing DUT configuration

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
platform-configs/odroid-h4-plus.robot: added variables for dasharo compatibility testing

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
platform-configs/odroid-h4-plus.robot: added keywords for dasharo security and dasharo stability tests

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
lib/power-after-fail-lib.robot: updated to use new RTE commands for power switching

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
dasharo-compatibility/esp-scanning.robot: changed keyword looking for DTS

In ESP005.001 when looking for Dasharo Tools Suite boot entry the test
would try to match a string " Dasharo Tools Suite on (* "this did not work
in case of Odroid's firmware because entries are shown as "ENTRY (on DRIVE)".

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
platform-configs/odroid-h4-plus.robot: Added variables for Dasharo stability tests

Signed-off-by Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

macpijan commented:
@mysteriousCzrsX Please wrap commit messages:

![image](https://github.com/user-attachments/assets/978ceda6-f79f-44f1-b496-4e52eafb26e6)


If you install pre-commit, it should force to you to do so: https://github.com/Dasharo/open-source-firmware-validation/blob/main/docs/contributing.md#pre-commit-and-ci-checks

macpijan commented on dasharo-compatibility/esp-scanning.robot:
```diff
@@ -86,7 +86,7 @@ ESP005.001 ESP Scan detects Dasharo Tools Suite
     Power On
     ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
-    Should Contain Match    ${boot_menu}    Dasharo Tools Suite on (*
+    Should Contain Match    ${boot_menu}    Dasharo Tools Suite *
```
I think the correct fix would be: `(on `. On other platforms it also looks like this. It is coded this way and has not changed in a while: https://github.com/Dasharo/edk2/blob/dasharo/MdeModulePkg/Library/UefiBootManagerLib/BmBoot.c#L2536

macpijan replied:
@mysteriousCzrsX I would appreciate a response and link to the given commit if you believe you have solved the given thread

mysteriousCzrsX replied:
Fixed: [here](https://github.com/Dasharo/open-source-firmware-validation/compare/dcef3d53c95da99dde101a02257da929afab7ed3..a3e6a42fbe6952e15f97c1cf31c461360f74dddc#diff-c8f7a5cdc7c64ba736049ef48cf813ae918ab342659418e21d720981c53f903aL89-R89)

macpijan commented on lib/power-after-fail-lib.robot:
```diff
@@ -17,8 +17,8 @@ Simulate Power Failure
         Sonoff On
     ELSE
-        Rte Relay Toggle
+        Rte Psu Off
```
@mysteriousCzrsX This whole keyword can be reworked. You may leave only the content in the ELSE branch and update the comment. The abstraction of whether sonoff or relay is used to control power is already done in osfv_cli libraries since https://github.com/Dasharo/osfv-scripts/pull/78

mysteriousCzrsX replied:
Fixed: [here](https://github.com/Dasharo/open-source-firmware-validation/pull/661/commits/87dd90899ebae2a2978e2d4682ef9d20629541d4)

macpijan commented:
@mysteriousCzrsX In platform cofnig both of these should say `Telnet` for this boad:

```
${INITIAL_DUT_CONNECTION_METHOD}=       pikvm
${DUT_CONNECTION_METHOD}=               pikvm
```


This platform has regular serial port we can use, no reason for us to use PiKVM for input in tests here. I think I had done it in https://github.com/Dasharo/open-source-firmware-validation/pull/659/files as well when testing changes on odroid.

macpijan commented on platform-configs/odroid-h4-plus.robot:
```diff
@@ -15,29 +15,42 @@ ${WIFI_CARD}=                           ${TBD}
+${DCU_UUID_SUPPORT}=                    ${TRUE}
+${DCU_SERIAL_SUPPORT}=                  ${TRUE}
+${CUSTOM_LOGO_SUPPORT}=                 ${TRUE}
+${DCU_SUPPORTED_BOOLEAN_SMMSTORE_VARIABLE}=    ${EMPTY}
```
Another proof you are not using pre-commit. That would be automatically aligned if you would. Please install it in this repo (`pre commit install`).

mysteriousCzrsX replied:
Fixed: [here](https://github.com/Dasharo/open-source-firmware-validation/pull/661/commits/a3e6a42fbe6952e15f97c1cf31c461360f74dddc)

mysteriousCzrsX committed:
platform-configs/odroid-h4-plus.robot: fixed formatting

Fixed formatting to comply with style guidelines

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
platform-configs/odroid-h4-plus.robot: changed connection method to Telnet

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
lib/power-after-fail-lib.robot: removed sonoff power control

Switching between using sonoff or RTE relay is already handled by osfv_cli libraries.

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
platform-configs/odroid-h4-plus.robot: added variables for dasharo-security

Added missing variables to enable verified boot and measured boot tests

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
keywords.robot: modified keywords for Linux display tests

Modified a command used for listing HDMI and DP ports to list all available ports.
Output of the command is split on a newline and all lines are checked so
if any of the available ports is connected the test will pass.

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

mysteriousCzrsX committed:
lib/options/uefi-setup-menu.robot: removed unnecessary argument

Passing an argument to a Power Cycle On keyword caused an exception
since this keyword does not require any arguments.

Signed-off-by: Cezary Wieczorkowski <cezary.wieczorkowski@3mdeb.com>

macpijan commented:
I am going to merge this current state, we can bring more updates in a followup if needed.
